trace-forward/src/Trace/Forward/Network/Acceptor.hs

```diff
@@ -0,0 +1,171 @@
+  ( listenToForwarder
+  -- | Export this function for Mux purpose.
+  , acceptTraceObjects
+  , acceptTraceObjectsInit
+  ) where
+
+import           Codec.CBOR.Term (Term)
+import qualified Codec.Serialise as CBOR
+import           Control.Concurrent.Async (async, wait, waitAnyCancel)
+import           Control.Monad (void)
+import qualified Data.ByteString.Lazy as LBS
+import           Data.Functor ((<&>))
+import           Data.IORef (readIORef)
+import           Data.Typeable (Typeable)
+import           Data.Void (Void)
+import           Ouroboros.Network.Mux (MiniProtocol (..), MiniProtocolLimits (..),
+                                        MiniProtocolNum (..), MuxMode (..),
+                                        OuroborosApplication (..), MuxPeer (..),
+                                        RunMiniProtocol (..),
+                                        miniProtocolLimits, miniProtocolNum, miniProtocolRun)
+import           Ouroboros.Network.Driver.Limits (ProtocolTimeLimits)
+import           Ouroboros.Network.Driver.Simple (runPeer)
+import           Ouroboros.Network.ErrorPolicy (nullErrorPolicies)
+import           Ouroboros.Network.IOManager (IOManager)
+import           Ouroboros.Network.Snocket (Snocket, localAddressFromPath, localSnocket)
+import           Ouroboros.Network.Socket (AcceptedConnectionsLimit (..),
+                                           SomeResponderApplication (..),
+                                           cleanNetworkMutableState, newNetworkMutableState,
+                                           nullNetworkServerTracers, withServerNode)
+import           Ouroboros.Network.Protocol.Handshake.Codec (cborTermVersionDataCodec,
+                                                             noTimeLimitsHandshake)
+import           Ouroboros.Network.Protocol.Handshake.Unversioned (UnversionedProtocol (..),
+                                                                   UnversionedProtocolData (..),
+                                                                   unversionedHandshakeCodec,
+                                                                   unversionedProtocolDataCodec)
+import           Ouroboros.Network.Protocol.Handshake.Type (Handshake)
+import           Ouroboros.Network.Protocol.Handshake.Version (acceptableVersion,
+                                                               simpleSingletonVersions)
+import           Ouroboros.Network.Util.ShowProxy (ShowProxy(..))
+
+import qualified Trace.Forward.Protocol.Acceptor as Acceptor
+import qualified Trace.Forward.Protocol.Codec as Acceptor
+import           Trace.Forward.Protocol.Type
+import           Trace.Forward.Queue (getTraceObjects)
+import           Trace.Forward.Configuration (AcceptorConfiguration (..), HowToConnect (..))
+
+listenToForwarder
+  :: (CBOR.Serialise lo,
+      ShowProxy lo,
+      Typeable lo)
+  =>  IOManager
+  -> AcceptorConfiguration lo
+  -> ([lo] -> IO ())
+  -> (NodeInfo -> IO ())
```

Comment:
I cannot find where production actions are defined.   Is the acceptor process developed in a separate branch?

Reply:
Production actions are defined in `cardano-tracer` (a new application that uses _acceptor's side_ of `trace-forward` library). But `cardano-tracer` is still in another branch.

Reply:
I see, thanks for clarification. Could you share a link to them?